fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryUnionPlanBase.java

```diff
@@ -159,8 +159,9 @@ public int hashCodeWithoutChildren() {
     }
 
     @Override
-    public int planHash() {
-        return PlanHashable.planHash(getQueryPlanChildren()) + (reverse ? 1 : 0);
+    public int planHash(PlanHashKind hashKind) {
+        // TODO: Should we include the concrete subtype type?
+        return PlanHashable.planHash(hashKind, getQueryPlanChildren()) + (reverse ? 1 : 0);
```

Comment:
I think this builds down the the general problem whether the kind of thing should be identifiable, i.e. should the class be incorporated here?